fhirpipe_clean/extract/sql.py

```diff
@@ -0,0 +1,292 @@
+            joins = joins.union(joins_child)
+
+    elif isinstance(tree, list) and len(tree) > 0:
+        for t in tree:
+            cols_children, joins_children = find_cols_and_joins(t)
+            columns += cols_children
+            joins = joins.union(joins_children)
+
+    return columns, joins
+
+
+def build_graph(joins):
+    """
+    Transform a join info into SQL fragments and parse the graph of join dependency
+    Input:
+        [(<type_of_join>, "<owner>.<table>.<col>=<owner>.<join_table>.<join_col>"), ... ]
+    Return:
+        [(
+            "<join_table>",
+            "<table>.<col> = <join_table>.<join_col>"
+        ), ... ],
+        graph of dependency of type DependencyGraph
+    """
+    print("build graph")
+    print(joins)
+    joins_elems = dict()
+    graph = DependencyGraph()
+    for join in joins:
+        join_source, join_target = join
+
+        # Get table names
+        target_table = get_table_name(join_target)
+        source_table = get_table_name(join_source)
+
+        # Add the join in the join_graph
+        source_node = graph.get(source_table)
+        target_node = graph.get(target_table)
+        source_node.connect(target_node)
+    return graph
+
+
+def build_squash_rule(node, table_col_idx):
+    """
+    Using the dependency graph of the joins on the tables (accessed through the
+    head node), regroup (using the id) the columns which should be squashed (ie
+    those accessed through a OneToMany join)
+    :param node: the node of the source table (which can be relative in recursive calls)
+    :param table_col_idx: a dict [table_name]: list of idx of cols in the SQL response
+        which come from this table
+    :return: [
+        (idx cols for source_table),
+        [
+            (idx cols for join OneToMany n1, []),
+            (idx cols for join OneToMany n2, []),
+            ...
+        ]
+    ]
+    """
+    # We refer the col indices of the table
+    unifying_col_idx = table_col_idx[node.name]
+
+    # Now parse the col indices for each table joined with a OneToMany
+    child_rules = []
+    for join_node in node.connections:
+        # print(node.name, "-<=", join_node.name)
+        child_rules.append(build_squash_rule(join_node, table_col_idx))
+    return [unifying_col_idx, child_rules]
+
+
+def get_connection(connection_type: str = None):
+    """
+    Return a sql connexion depending on the configuration provided in
+    config.yml (see root of the project)
+    It should be used in a context environment (with get_connection(c) as ...)
+
+    args:
+        connection_type (str): a string like "postgre", "oracle". See your
+            config file for available values
+
+    return:
+        a sql connexion
+    """
+    sql_config = fhirpipe_clean.global_config.sql.to_dict()
+    if connection_type is None:
+        connection_type = sql_config["default"]
+    connection = sql_config[connection_type]
+    try:
+        lib = eval(connection["lib"])
```

Comment:
Based on the l 251 of this file and the fact the eval is really not loved in the python community, we could make a switch like in l 251 when dealing with the LIMIT syntax, and remove the eval